src/main/webapp/resources/js/pages/projects/samples/linker/Linker.jsx

```diff
@@ -27,33 +27,79 @@ const CommandText = styled(Paragraph)`
  * @constructor
  */
 function Linker() {
+  const options = [
+    { label: i18n("Linker.fastq"), value: "fastq" },
+    { label: i18n("Linker.assembly"), value: "assembly" }
+  ];
+
+  //this will hold the base string of the script with the projects & samples.  The filetypes will be appended before displaying.
+  let scriptString = undefined;
+
+  let modal = undefined; 
+
+  const ModalContents = (
+    { command } 
+  ) => (
+    <>
+      {" "}
+      <Paragraph>{i18n("Linker.details")}</Paragraph>
+      <Text type="secondary">
+        <span dangerouslySetInnerHTML={{ __html: i18n("Linker.note") }} />
+      </Text>
+      <br/><br/>
+      <Text strong>{i18n("Linker.types")}</Text> <Checkbox.Group
+        options={options}
+        defaultValue={["fastq"]}
+        onChange={updateCommand}
+      />
+      <CommandText
+        className="t-cmd-text"
+        ellipsis={{ rows: 1 }}
+        copyable={ command }
+      >
+        {command}
+      </CommandText>
+      
+    </>
+  );
+
   function handleSampleIds(e) {
-    // Post data to get linker command.
+    // Post data to the server to get the linker command.
     const { detail } = e;
     getNGSLinkerCode(detail).then(({ data }) => {
-      Modal.success({
+      scriptString = data;
```

Comment:
You can rename a variable when you are destructuring the object, so:

```
 getNGSLinkerCode(detail).then(({ data : scriptString }) => {
```
would give the same result.

Reply:
So this wasn't just renaming the variable.  That variable is at https://github.com/phac-nml/irida/pull/638/files#diff-383b10c14b55cb1fcd2260780fbcf323R36 to hold the base of the script string so when it gets updated it doesn't have to repeat the server call.  See how it's building the string to display at https://github.com/phac-nml/irida/pull/638/files#diff-383b10c14b55cb1fcd2260780fbcf323R97.

Let me know if there's a different way to go about this.